core/trino-main/src/main/java/io/trino/execution/scheduler/FaultTolerantStageScheduler.java

```diff
@@ -397,7 +398,17 @@ private void startTask(int partition, NodeAllocator.NodeLease nodeLease, MemoryR
         int attemptId = getNextAttemptIdForPartition(partition);
 
         ExchangeSinkHandle sinkHandle = partitionToExchangeSinkHandleMap.get(partition);
-        ExchangeSinkInstanceHandle exchangeSinkInstanceHandle = sinkExchange.instantiateSink(sinkHandle, attemptId);
+        ExchangeSinkInstanceHandle exchangeSinkInstanceHandle = null;
+        try {
+            exchangeSinkInstanceHandle = sinkExchange.instantiateSink(sinkHandle, attemptId).get(1, TimeUnit.SECONDS);
```

Comment:
Reason for timeout to be different from `DeduplicatingDirectExchangeBuffer`?

Reply:
This class will be gone. DOes not matter